source/dsl/GitHubDSL.ts

```diff
@@ -1,9 +1,27 @@
+import { GitCommit } from "./Commit"
+
+/** A GitHub specific implmentation of a git commit */
+export interface GitHubCommit {
+  /** The raw commit metadata */
+  commit: GitCommit,
+  /** The SHA for the commit */
+  sha: string,
+  /** the url for the commit on GitHub */
+  url: string,
+  /** The GitHub user who wrote the code */
+  author: GitHubUser,
+  /** The GitHub user who shipped the code */
+  committer: GitHubUser,
+  /** An array of parent commit shas */
+  parents: Array<any>
```

Comment:
Generally in TS any approach's ok as long as there's consistency. My personal preference is generic type for readability, you're getting type of actual object (`Array<>` to inner generic type `string`) left to right order by reading it. Also it gives consistency like 

`class MyArray extends Array<string> { ... }`

extending form can't use array of element type.

I also remember I've told to use generic type as preferred, but I don't recall where and when.